src/anomalib/engine/engine.py

```diff
@@ -155,8 +156,12 @@ def __init__(
         if self.task == TaskType.SEGMENTATION:
             self.pixel_metric_names = pixel_metrics if pixel_metrics is not None else ["AUROC", "F1Score"]
 
+        self._model: AnomalyModule | None = None
```

Comment:
What is the difference between `self.model` and `self._model`?

Reply:
`self._model` is a field. I created a getter/setter `self.model` to move the `self._setup_trainer(model)` call to the setter and keep the logic simpler.